Periodics/PeriodicsLibrary/Catalogue.cs

```diff
@@ -0,0 +1,20 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace PeriodicsLibrary
+{
+    public class Catalogue : IEnumerable<Edition>
+    {
+        private List<Edition> allEditions;
+        public int Count { get => allEditions.Count; }
```

Comment:
Можно еще меньше инфраструктурного кода
```suggestion
        public int Count => allEditions.Count;
```